service/internal/testcomponents/stateful_component.go

```diff
@@ -0,0 +1,50 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testcomponents // import "go.opentelemetry.io/collector/service/internal/testcomponents"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+type StatefulComponent interface {
```

Comment:
I know that it is internal package, but do you need to expose this public? And more than that, do you need it at all?

Reply:
I believe it will be useful when validating complex component graphs, because each component can be treated as an interface. But it is not needed in this PR so I've removed it.